tempelate_engine/tempelate_engine.js

```diff
@@ -0,0 +1,89 @@
+/*
+*========usage=========
+*body = {
+*  name: "Jim",
+*  id: "123",
+*  token: "abc",
+*  tasks: ["enter", "exit"]
+*}
+*
+*tempelate = {
+*  username: "{{payload.name}}",
+*  clientId: "{{payload.id}}",
+*  accessToken: "{{payload.token}}",
+*  task: "payload.tasks.0"
+*}
+*
+*context = {payload: body}
+*
+*engine = new TempelateEngine(tempelate, context);
+*engine.render();
+*engine.getTempelate() // returns rendered tempelete
+*
+*/
+
+class TempelateEngine {
```

Comment:
I hate to nitpick on spelling, but the "e" in the middle shouldn't be there: "Template" :)